src/components/progressCircular/js/progressCircularDirective.js

```diff
@@ -103,22 +108,34 @@ function MdProgressCircularDirective($$rAF, $window, $mdProgressCircular, $mdThe
       startIndeterminateAnimation();
     }
 
-    scope.$watchGroup(['value', 'mdMode'], function(newValues, oldValues) {
+    scope.$watchGroup(['value', 'mdMode', function() {
+      return node.disabled;
+    }], function(newValues, oldValues) {
       var mode = newValues[1];
+      var isDisabled = newValues[2];
+      var wasDisabled = oldValues[2];
 
-      if (mode !== MODE_DETERMINATE && mode !== MODE_INDETERMINATE) {
-        mode = MODE_INDETERMINATE;
-        attrs.$set('mdMode', mode);
+      if (isDisabled !== wasDisabled) {
```

Comment:
`wasDisabled` will be always different from the new `isDisabled`. So this check seems to be unnecessary?